core/drag/drag-event.js

```diff
@@ -0,0 +1,343 @@
+            }
+        },
+
+        _dropEffect: {
+            value: null
+        },
+
+        dropEffect: {
+            set: function (effect) {
+                if (
+                    effect &&
+                    DataTransfer.ALLOWED_EFFECTS.indexOf(effect) > -1 &&
+                    DataTransfer.isDropEffectAllowed(
+                        effect, this.effectAllowed
+                    )
+                ) {
+                    this._dropEffect = effect;
+                } else {
+                    this._dropEffect = null;
+                }
+            },
+            get: function () {
+                if (!this._dropEffect) {
+                    var index;
+
+                    if (
+                        this.effectAllowed === DataTransfer.All ||
+                        this.effectAllowed.startsWith('c')
+                    ) {
+                        this._dropEffect = DataTransfer.Copy;
+                    } else if ((index = DataTransfer.ALLOWED_EFFECTS.indexOf(
+                        this.effectAllowed)) > -1
+                    ) {
+                        this._dropEffect = DataTransfer.ALLOWED_EFFECTS[index];
+                    } else {
+                        this._dropEffect = DataTransfer.Link;
+                    }
+                }
+
+                return this._dropEffect;
+            }
+        },
+
+        _effectAllowed: {
+            value: null
+        },
+
+        effectAllowed: {
+            set: function (effect) {
+                if (DataTransfer.ALLOWED_DROP_EFFECTS.indexOf(effect) > -1) {
+                    this._effectAllowed = effect;
+                }
+            },
+            get: function () {
+                return this._effectAllowed || DataTransfer.All;
+            }
+        },
+
+        files: {
+            value: null
+        },
+
+        items: {
+            value: null
+        },
+
+        types: {
+            value: null
+        },
+
+        dragTarget: {
+            value: null
+        },
+
+        _candidateDropTargets: {
+            value: null,
+            enumerable: false
+        },
+
+        candidateDropTargets: {
+            get: function () {
+                return this._candidateDropTargets ||
+                    (this._candidateDropTargets = new Set());
+            }
+        },
+
+        dragabblePlaceholderStrategy: {
+            value: 'hidden'
+        },
+
+        clearData: {
+            value: function () {
+                return this._data.clear();
+            }
+        },
+
+        getData: {
+            value: function (key) {
+                return this._data.get(key);
+            }
+        },
+
+        hasData: {
+            value: function (key) {
+                return this._data.has(key);
+            }
+        },
+
+        setData: {
+            value: function (key, value) {
+                return this._data.set(key, value);
```

Comment:
Add specified mime type for objects `application/object`

add sugar method `draggedObject`